src/filesystem_littlefs.c

```diff
@@ -0,0 +1,517 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <time.h>
+#include <string.h>
+#include <sys/iosupport.h>
+#include <sys/stat.h>
+#include <sys/syslimits.h>
+#include "filesystem.h"
+#include "filesystem_littlefs.h"
+#include "littlefs.h"
+
+#define RESULT result < 0 ? -1 : 0;
```

Comment:
Are you sure the `result` will always get expanded correctly into this ternary operator? I'd suggest adding parentheses at the very least...

Reply:
It's used only as `return RESULT;` and adding anything before or after this macro would break it on compile time.